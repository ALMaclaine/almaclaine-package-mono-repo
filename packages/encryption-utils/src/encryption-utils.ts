import {hash, compare} from 'bcrypt';

export async function encryptPassword(password: string, rounds: number = 64000) {
    return hash(password, rounds);
}

export async function comparePassword(providedPassword: string, actualPassword: string) {
    return compare(providedPassword, actualPassword);
}
