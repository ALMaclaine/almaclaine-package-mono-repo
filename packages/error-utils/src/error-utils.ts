export class ALMError extends Error {
    constructor(message: string, public readonly type: string) {
        super(message);
    }
}
