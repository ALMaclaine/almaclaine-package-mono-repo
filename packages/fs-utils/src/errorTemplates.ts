export const doesNotExistError = (path: string): string => `Provided path ${path} does not exist: ${path}.`;

export const alreadyExistsError = (path: string): string => `Provided file/directory already exists: ${path}.`;

export const isNotDirectoryError = (path: string): string => `Provided path ${path} is not a directory.`;

export const isDirectoryError = (path: string): string => `Provided path ${path} is a directory.`;

export const isNotJsonFilError = (path: string): string => `Provided path ${path} is not a json file. Extension must be .json`;

