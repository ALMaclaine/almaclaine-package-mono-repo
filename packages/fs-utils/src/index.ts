export * from './fs-utils';
export * from './errorTemplates';
