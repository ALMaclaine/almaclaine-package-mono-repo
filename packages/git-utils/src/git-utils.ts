import {execSync} from 'child_process';

export function commitPackage(pack: string, desc: string): void {
    console.log(`Committing Project: ${pack}.`)
    const stripped = desc.replace(/"/g, '');
    let out = execSync('git add .').toString('utf-8').trim();
    console.log(out);
    out = execSync(`git commit -m "${pack}: ${stripped}."`).toString('utf-8').trim();
    console.log(out);
}
