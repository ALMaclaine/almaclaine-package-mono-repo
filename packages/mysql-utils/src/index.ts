export * from './mysql-utils';
export type {ConnectionInfo} from './types';
