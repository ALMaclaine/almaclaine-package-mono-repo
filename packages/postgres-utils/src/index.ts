export * from './postgres-utils';
export * from './ConnectionManager';
