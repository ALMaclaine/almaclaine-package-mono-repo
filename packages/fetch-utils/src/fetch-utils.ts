const fetch = require('cross-fetch');
import {ALMError} from "@almaclaine/error-utils";

export const ErrorTypes = {
    FetchFailed: 'FetchFailed'
}

export class FetchFailedError extends ALMError {
    constructor(message: string) {
        super(message, ErrorTypes.FetchFailed);
    }
}

export async function almFetch(url: string, options: {[key: string]: any} = {}) {
    const res = await fetch(url, options);
    if(!res.ok) {
        throw new FetchFailedError(`Fetch failed with status code: ${res.status}`);
    }
    return res;
}
