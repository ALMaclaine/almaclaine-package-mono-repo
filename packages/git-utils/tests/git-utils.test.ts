import * as gitUtils from '../src';

describe('utils/git-utils', () => {
    it('Has Correct API', () => {
        const keys = Object.keys(gitUtils);
        expect(keys.length).toBe(1);
        expect(keys).toMatchSnapshot();
    });
});
