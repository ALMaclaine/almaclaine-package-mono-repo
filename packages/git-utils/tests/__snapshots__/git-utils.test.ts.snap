// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/git-utils Has Correct API 1`] = `
Array [
  "commitPackage",
]
`;
